src/io/csv/csv-reader.cu

```diff
@@ -770,11 +787,12 @@ __global__ void convertCsvToGdf(
 					break;
 			}
 
-				// // set the valid bitmap - all bits were set to 0 to start
-				int bitmapIdx 	= whichBitmap(rec_id);  	// which bitmap
-				int bitIdx		= whichBit(rec_id);		// which bit - over an 8-bit index
-				setBit(valid[col]+bitmapIdx, bitIdx);		// This is done with atomics
+			// set the valid bitmap - all bits were set to 0 to start
+			int bitmapIdx 	= whichBitmap(rec_id);  	// which bitmap
+			int bitIdx		= whichBit(rec_id);		// which bit - over an 8-bit index
+			setBit(valid[col]+bitmapIdx, bitIdx);		// This is done with atomics
 
+			atomicAdd((unsigned long long int*)&num_valid[col],(unsigned long long int)1);
```

Comment:
It looks like all threads are atomically incrementing a global variable. Consider using a shared memory counter per block, and then writing that to global memory at the end.

Reply:
I agree there are some better ways to do this.  There are some upcoming tasks to break the code into primitives, null_count being one of them, and update for it is performed.